vowpalwabbit/autolink.cc

```diff
@@ -34,17 +34,20 @@ void predict_or_learn(autolink& b, LEARNER::single_learner& base, example& ec)
   ec.indices.pop();
 }
 
-LEARNER::base_learner* autolink_setup(arguments& arg)
+LEARNER::base_learner* autolink_setup(VW::config::options_i& options, vw& all)
 {
   free_ptr<autolink> data = scoped_calloc_or_throw<autolink>();
-  if (arg.new_options("Autolink")
-      .critical("autolink", data->d, "create link function with polynomial d").missing())
+  VW::config::option_group_definition new_options("Autolink");
+  new_options.add(VW::config::make_typed_option("autolink", data->d).keep().help("create link function with polynomial d"));
+  options.add_and_parse(new_options);
+
+  if (!options.was_supplied("autolink"))
     return nullptr;
 
-  data->stride_shift = arg.all->weights.stride_shift();
+  data->stride_shift = all.weights.stride_shift();
 
   LEARNER::learner<autolink,example>& ret =
-    init_learner(data, as_singleline(setup_base(arg)), predict_or_learn<true>, predict_or_learn<false>);
+    init_learner(data, as_singleline(setup_base(*all.options, all)), predict_or_learn<true>, predict_or_learn<false>);
```

Comment:
It's somewhat inelegant to pass an argument and a field of the same argument.

Reply:
I agree it is inelegant, but I think it is preferable to just the all object no? It is more explicit about parameters. It is unfortunate that the option object must be a field of all still

Reply:
I'll make the change